object.c

```diff
@@ -3184,35 +3184,71 @@ rb_opts_exception_p(VALUE opts, int default_value)
 
 /*
  *  call-seq:
- *     Integer(arg, base=0, exception: true)    -> integer or nil
- *
- *  Converts <i>arg</i> to an Integer.
- *  Numeric types are converted directly (with floating point numbers
- *  being truncated).  <i>base</i> (0, or between 2 and 36) is a base for
- *  integer string representation.  If <i>arg</i> is a String,
- *  when <i>base</i> is omitted or equals zero, radix indicators
- *  (<code>0</code>, <code>0b</code>, and <code>0x</code>) are honored.
- *  In any case, strings should consist only of one or more digits, except
- *  for that a sign, one underscore between two digits, and leading/trailing
- *  spaces are optional.  This behavior is different from that of
- *  String#to_i.  Non string values will be converted by first
- *  trying <code>to_int</code>, then <code>to_i</code>.
- *
- *  Passing <code>nil</code> raises a TypeError, while passing a String that
- *  does not conform with numeric representation raises an ArgumentError.
- *  This behavior can be altered by passing <code>exception: false</code>,
- *  in this case a not convertible value will return <code>nil</code>.
- *
- *     Integer(123.999)    #=> 123
- *     Integer("0x1a")     #=> 26
- *     Integer(Time.new)   #=> 1204973019
- *     Integer("0930", 10) #=> 930
- *     Integer("111", 2)   #=> 7
- *     Integer(" +1_0 ")   #=> 10
- *     Integer(nil)        #=> TypeError: can't convert nil into Integer
- *     Integer("x")        #=> ArgumentError: invalid value for Integer(): "x"
- *
- *     Integer("x", exception: false)        #=> nil
+ *    Integer(object, base = 0, exception: true) -> integer or nil
+ *
+ *  Returns an integer converted from +object+.
+ *
+ *  Tries to convert +object+ to an integer
+ *  using +to_int+ first and +to_i+ second;
+ *  exception: when +object+ is a string, does _not_ call String#to_i
```

Comment:
There are exceptions for integer, float, and nil, in addition to string.  Maybe just end the paragraph with `+to_i+ second (see below for exceptions).`

Behavior for `nil` argument was previously given, and should not be dropped.  I realize this goes against the standard logic for not documenting cases that raise exceptions, but this behavior is explicit and goes against the `to_int`/`to_i` behavior you are describing, so I think it should be mentioned.

Reply:
Both issues fixed, I think.